sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeInvoicesFromUrl.java

```diff
@@ -111,15 +112,54 @@ public static void main(final String[] args) throws IOException {
                         invoiceTotal, invoiceTotalField.getConfidence());
                 }
             }
-        }
 
-        // The .subscribe() creation and assignment is not a blocking call. For the purpose of this example, we sleep
-        // the thread so the program does not end before the send operation is complete. Using .block() instead of
-        // .subscribe() will turn this into a synchronous call.
-        try {
-            TimeUnit.MINUTES.sleep(1);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
+            FormField invoiceItemsField = recognizedFields.get("Items");
+            if (invoiceItemsField != null) {
+                System.out.printf("Invoice Items: %n");
+                if (FieldValueType.LIST == invoiceItemsField.getValue().getValueType()) {
+                    List<FormField> invoiceItems = invoiceItemsField.getValue().asList();
+                    invoiceItems.stream()
+                        .filter(invoiceItem -> FieldValueType.MAP == invoiceItem.getValue().getValueType())
+                        .map(formField -> formField.getValue().asMap())
+                        .forEach(formFieldMap -> formFieldMap.forEach((key, formField) -> {
+                            if ("Description".equals(key)) {
+                                if (FieldValueType.STRING == formField.getValue().getValueType()) {
+                                    String name = formField.getValue().asString();
+                                    System.out.printf("Description: %s, confidence: %.2fs%n",
+                                        name, formField.getConfidence());
+                                }
+                            }
+                            if ("Quantity".equals(key)) {
+                                if (FieldValueType.FLOAT == formField.getValue().getValueType()) {
+                                    Float quantity = formField.getValue().asFloat();
+                                    System.out.printf("Quantity: %f, confidence: %.2f%n",
+                                        quantity, formField.getConfidence());
+                                }
+                            }
+                            if ("Unit".equals(key)) {
+                                if (FieldValueType.FLOAT == formField.getValue().getValueType()) {
+                                    Float unit = formField.getValue().asFloat();
```

Comment:
how is this working for you? the service is returning:
```
"Unit": {
  "type": "number",
  "text": "hours",
  "boundingBox": [
    965,
    1138,
    1043,
    1139,
    1043,
    1164,
    965,
    1162
  ],
  "page": 1,
  "confidence": 0.895
},
```

in .NET parsing to Float breaks the code.

Reply:
So since the type is number the check on L140 would never be true and the value will not be printed. I dont think I have this code for line items checking in the tests yet.

Reply:
Also, just checked the sample_invoice.jpg currently is returning just these fields
```java
"Amount" -> {FormField@4691} 
"Description" -> {FormField@4693} 
"Quantity" -> {FormField@4695} 
"UnitPrice" -> {FormField@4697} 
```

Reply:
If the service starts finding that field at some point will the sample break? Since you're only showing some of the line item fields, maybe just remove that one?

Also, I saw that .NET added a small comment in their PR saying that only some of the fields were shown and then provided the aka.ms link with all the fields available. Would it make sense for java to do that too?

Reply:
>If the service starts finding that field at some point will the sample break?
If the service returns it in the same format as Mariana mentioned then yes, that would break the sample.

>Also, I saw that .NET added a small comment in their PR saying that only some of the fields were shown and then provided the aka.ms link with all the fields available. Would it make sense for java to do that too?
yeah, I actually have that aka.ms link above https://github.com/Azure/azure-sdk-for-java/pull/20369/files#diff-e5752bbeac74fff2958d578f40f112e6dc73e518cb673b0d2e422801b1a9a181L30. 
I think it should be fine at the start so that if the sample is missing any other reference fields that should be good to find.